vernac/vernacentries.ml

```diff
@@ -2211,18 +2172,69 @@ let with_def_attributes ~atts f =
   if atts.DefAttributes.program then Obligations.check_program_libraries ();
   f ~atts
 
+(** A global default timeout, controlled by option "Set Default Timeout n".
+    Use "Unset Default Timeout" to deactivate it (or set it to 0). *)
+
+let default_timeout = ref None
+
+(* Timeout *)
+let vernac_timeout ?timeout (f : 'a -> 'b) (x : 'a) : 'b =
+  match !default_timeout, timeout with
+  | _, Some n
+  | Some n, None ->
+    Control.timeout n f x Timeout
+  | None, None ->
+    f x
+
+(* Fail *)
+exception HasNotFailed
+exception HasFailed of Pp.t
+
+let test_mode = ref false
+
+(* XXX STATE: this type hints that restoring the state should be the
+   caller's responsibility *)
```

Comment:
I don't understand this comment.

Reply:
The comment was here before; I am not sure what it means anymore tho; see #9855 however that one is work in progress.

Reply:
I'd say let's leave the comment as it was for this PR.